### ISSUE TITLE
data._entities elements amount is not correct

### ISSUE THREAD

**Issue by scottming** [DivvyPayHQ/absinthe_federation#32]
While I was exploring absinthe_federation to try to solve the good first issue, I found another bug.

When you query this

```gql
query getAllProducts {
  allProducts {
    createdBy {
      email
      name
    }
    id
  }
}

```

it returns this error

```json
{
  "errors": [
    {
      "message": "Expected \"data._entities\" to contain 2 elements",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: Expected \"data._entities\" to contain 2 elements",
            "    at /Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/@apollo/gateway/dist/executeQueryPlan.js:222:27",
            "    at processTicksAndRejections (node:internal/process/task_queues:96:5)",
            "    at async executeNode (/Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/@apollo/gateway/dist/executeQueryPlan.js:132:17)",
            "    at async executeNode (/Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/@apollo/gateway/dist/executeQueryPlan.js:120:27)",
            "    at async executeNode (/Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/@apollo/gateway/dist/executeQueryPlan.js:103:40)",
            "    at async /Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/@apollo/gateway/dist/executeQueryPlan.js:29:35",
            "    at async /Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/@apollo/gateway/dist/index.js:126:38",
            "    at async execute (/Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/apollo-server-core/dist/requestPipeline.js:201:20)",
            "    at async processGraphQLRequest (/Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/apollo-server-core/dist/requestPipeline.js:138:28)",
            "    at async processHTTPRequest (/Users/scott.mingcrypto.com/Code/federation_poc/gateway/node_modules/apollo-server-core/dist/runHttpQuery.js:187:30)"
          ]
        }
      }
    }
  ],
  "data": {
    "allProducts": [
      {
        "createdBy": {
          "email": "support@apollographql.com",
          "name": null
        },
        "id": "apollo-federation"
      },
      {
        "createdBy": {
          "email": "support@apollographql.com",
          "name": null
        },
        "id": "apollo-studio"
      }
    ]
  }
}
```

The reason is the `localhost:4002` returns only one elements:

```gql
query ($representations: [_Any!]!) {
  _entities(representations: $representations) {
    ... on User {
      name
      email
    }
  }
}

```

```json
{
    "representations":
    [
        {
            "__typename": "User",
            "email": "support@apollographql.com"
        }
    ]
}
```

You can reproduce the bug with this [repository](https://github.com/scottming/federation_poc/commit/7a8650177a91c5723c2ea94c7616e96c68a479e7#diff-bd96b8670346da6614b182946066640cf52769888c3e9a4eeffd247504a8851eR25)

**Comment by scottming** [comment]
I don't quite understand why `Enum.uniq/2` is used instead of `Enum.map` [here](https://github.com/DivvyPayHQ/absinthe_federation/blob/e1371bc2776e6cb64ed17f3ac6ec3d9b63e627d8/lib/absinthe/federation/schema/entities_field.ex#L88)

**Comment by kdawgwilk** [comment]
I took a look at this yesterday and there is definitely something weird going on. @SvenW do you know why we need that `uniq_by` seems to be deduping other resolutions that we do want to resolve.  I have a failing test for this right now I can push up

**Comment by SvenW** [comment]
> I took a look at this yesterday and there is definitely something weird going on. @SvenW do you know why we need that `uniq_by` seems to be deduping other resolutions that we do want to resolve. I have a failing test for this right now I can push up

We can probably do that part differently. The reason, if I remember correctly, was to make sure we only get the dataloader for a specific entity once and populate that middleware with all ids we want to resolve. But we can do that uniq part in a different stage of course

**Comment by scottming** [comment]
@SvenW 
Can you push some test cases about this thing? Then I can try to implement it.

**Comment by kdawgwilk** [comment]
@scottming this should be fixed in 0.1.8 if you want to give it a try
